app/src/main/java/chat/rocket/android/chatroom/adapter/ChatRoomAdapter.kt

```diff
@@ -204,6 +218,42 @@ class ChatRoomAdapter(
         }
     }
 
+    private val actionAttachmentOnClickListener = object : ActionAttachmentOnClickListener {
+        override fun onActionClicked(action: Action) {
+            val temp = action as ButtonAction
+            if (temp.url != null && temp.isWebView != null) {
+                if (temp.isWebView!!) {
+                    //Open in a configurable sizable webview
+                    Toast.makeText(context, "Open in a configurable sizable webview", Toast.LENGTH_SHORT).show()
+                } else {
+                    //Open in chrome custom tab
+                    with(this) {
+                        val customTabsBuilder = CustomTabsIntent.Builder()
+                        customTabsBuilder.setToolbarColor(ResourcesCompat.getColor(context!!.resources, R.color.colorPrimary, context.theme))
+                        val customTabsIntent = customTabsBuilder.build()
+                        try {
+                            customTabsIntent.launchUrl(context, Uri.parse(temp.url!!))
+                        } catch (ex: Exception) {
+                            Timber.d(ex, "Unable to launch URL")
+                        }
+                    }
+                }
+            } else if (temp.message != null && temp.isMessageInChatWindow != null) {
+                if (temp.isMessageInChatWindow!!) {
+                    //Send to chat window
+                    temp.message.run {
+                        if (roomId != null) {
+                            presenter?.sendMessage(roomId, temp.message!!, null)
+                        }
+                    }
+                } else {
+                    //Send to bot but not in chat window
+                    Toast.makeText(context, "Send to bot but not in chat window", Toast.LENGTH_SHORT).show()
```

Comment:
Hard coded string. Please, remove and use the string from the `string.xml` files.